### DISCUSSION TITLE
as router.push was previously used

### DISCUSSION THREAD

**Discussion by hanchen007** [vuepress/ecosystem#269] [Q&A]
After the 2.0.0-rc.3 release, the sidebar component was rewritten in custom themes. How should the route transition be written now, as router.push was previously used?

**Comment by Mister-Hope** [comment]
`route.push` is always supported, so what's your confusion?

**Reply by hanchen007** [reply]
Using "route.push" in a custom theme should not affect the packing speed, right? Currently, it takes an hour to package 1000 md files, and the packing slows down as it progresses.

**Reply by Mister-Hope** [reply]
RouterLink should be avoid to use because of it's high cost by calling `router.resolve()` internally, there is nothing related with `router.push`.

In short you should try to avoid resolving routes in vuepress

**Reply by hanchen007** [reply]
Can use VPLink instead of RouterLink? RouterLink was used in the sidebar

**Reply by Mister-Hope** [reply]
Yes, VPLink is the component we recommended if the link has no active state or its active state is maintained outside the component.
